knowledge-base/fileselect-autofac.md

```diff
@@ -0,0 +1,88 @@
+---
+title: FileSelect ReadAsync Method Throws JsonException
+description: Learn how to resolve an exception about JSON serialization is attempting to deserialize an unexpected byte array when using the Telerik Blazor FileSelect.
+type: troubleshooting
+page_title: FileSelect ReadAsync Method Throws JsonException
+slug: fileselect-kb-autofac
+tags: telerik, blazor, fileselect, autofac, dependency injection
+ticketid: 1637702, 1634460, 1634313, 1632874
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>FileSelect for Blazor</td>
+        </tr>
+        <tr>
+            <td>Version</td>
+            <td>5.0.0 and above</td>
+        </tr>
+    </tbody>
+</table>
+
+
+## Description
+
+This knowledge base article deals with the following issues:
+
+* FileSelect component file stream `ReadAsync` method throws `Microsoft.JSInterop.JSException`
+* FileSelect file stream reading fails when using `builder.Host.UseServiceProviderFactory`
+* An exception occurs when trying to upload files through Blazor FileSelect with Autofac dependency injection enabled.
+
+
+## Error Message
+
+>warning Microsoft.JSInterop.JSException
+>
+> JsonException: JSON serialization is attempting to deserialize an unexpected byte array.
+>
+> StackTrace:
+>
+> at System.Threading.Tasks.ValueTask`1.get_Result()
+> 
+> at Telerik.Blazor.Components.FileSelect.Stream.FileInfoStream.&lt;ReadBytesAsync&gt;d__24.MoveNext()
+
+
+## Cause
+
+The problem can occurs when an Inversion of Control (IoC) container interferes with the SignalR hub operation. For example, we have received reports about Autofac:
+
+>caption Program.cs
+
+<div class="skip-repl"></div>
+
+````CS
+builder.Host.UseServiceProviderFactory(new AutofacServiceProviderFactory());
+````
+
+The problem does not occur in older FileSelect versions before 5.0.0, because the component used a slower legacy file management mechanism with `string` instead of `byte[]`.
```

Comment:
```suggestion
The problem does not occur in FileSelect versions before 5.0.0, because the component used a slower legacy file management mechanism with `string` instead of `byte[]`.
```